src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -267,8 +268,19 @@ protected internal int GetInvocationCounter()
         /// <returns>The next random number.</returns>
         protected internal BigInteger GetRandom()
         {
-            nonceData = Cryptography.Helper.Murmur128(nonceData, ProtocolSettings.Network);
-            return new BigInteger(nonceData, isUnsigned: true);
+            if (IsHardforkEnabled(Hardfork.HF_Aspidochelone))
+            {
+                Span<byte> buffer = stackalloc byte[nonceData.Length + sizeof(int)];
+                nonceData.AsSpan().CopyTo(buffer);
+                BinaryPrimitives.WriteInt32LittleEndian(buffer[nonceData.Length..], random_times++);
+                buffer = Cryptography.Helper.Murmur128(buffer, ProtocolSettings.Network);
```

Comment:
Why don't concat the protocolSetting in the nonce, and the calLNumber as a nonce, we can save some copies

Reply:
Good idea! Updated.

Reply:
I mean copy the `ProtocolSettings` here https://github.com/neo-project/neo/blob/a1255c7b9a5e6617c80e091c16579f695b475f50/src/neo/SmartContract/ApplicationEngine.cs#L160

Reply:
In fact we don't need it. We use it as the seed before because we need a seed.